MCprep_addon/materials/generate.py

```diff
@@ -958,6 +958,27 @@ def set_saturation_material(mat):
 		sat_node.mute = not bool(saturate)
 		sat_node.hide = not bool(saturate)
 
+def create_node(tree_nodes, node_type):
+	"""Function to create node"""
+	if node_type == 'ShaderNodeMixRGB': # For MixRGB in 3.4 become Legacy 
+		if util.min_bv((3, 4, 0)):
+			node = tree_nodes.new('ShaderNodeMix')
+			node.data_type = 'RGBA'
+			inputs = [0,6,7]
+			outputs = [2]
+		else:
+			node = tree_nodes.new('ShaderNodeMixRGB')
+			inputs = [0,1,2]
+			outputs = [0]
+		sockets= {
+			"inputs": inputs,
+			"outputs": outputs
+		}
```

Comment:
I need to live test a little, but this feels a bit funny to me. I think it's important that a function generally has the same return structure regardless of how it runs (here, sometimes it returns one value, sometimes two, which makes it unclear what you get back down the line).

I'll probably propose an edit later today unless you beat me to it.

Did the socket _names_ change, or just the order? If the names we need to use are the same

Reply:
For the sockets part, I understand. It's quite odd for me sometimes returns one, two.
The new Mix changes the socket index and the name. If you use the socket name, you will get the result of a float mix instead of RGBA. 
![image](https://user-images.githubusercontent.com/61040487/206229425-4ef7377b-5291-4859-8a2a-d22275b5d0a4.png)

Reply:
This may be the C++/Rust part of my brain acting up, but couldn't we just either:
A. Always return a tuple regardless of the amount of objects, where the first item is the node and the second object depending on the version can be the sockets 
B. Use a small wrapper class to handle this situation

Knowing how CPython works though, the C++/Rust part of my brain is screaming "unnecessary copy!" as I write this

Reply:
Maybe we could split the sockets part out to a different function like `get_node_socket("ShaderMixNode", "outputs",1)` 
It will do versioning for the socket of that node type seperate.